client/scripts/views/components/new_thread_form.ts

```diff
@@ -0,0 +1,386 @@
+    let formBodyMarkdown;
+    if (state.quillEditorState.markdownMode) {
+      formBodyMarkdown = quill.getText();
+    } else {
+      formBodyDelta = quill.getContents();
+    }
+
+    const discardedDraft = app.user.discussionDrafts.store
+      .getByCommunity(app.activeId())
+      .filter((d) => d.id === fromDraft)[0];
+    let discardedDelta;
+    let discardedMarkdown;
+    try {
+      discardedDelta = new Delta(JSON.parse(discardedDraft.body));
+    } catch {
+      discardedMarkdown = discardedDraft.body;
+    }
+    const titleIsChanged = discardedDraft.title
+      && (titleInput as HTMLInputElement).value !== discardedDraft.title;
+    const bodyIsChanged = formBodyDelta
+      ? !_.isEqual(formBodyDelta, discardedDelta)
+      : formBodyMarkdown
+        ? formBodyMarkdown !== discardedMarkdown
+        : false;
+    if (bodyIsChanged || titleIsChanged) {
+      confirmed = await confirmationModalWithText(overwriteDraftMsg)();
+    }
+  } else if (quill.getLength() > 1) {
+    confirmed = await confirmationModalWithText(overwriteDraftMsg)();
+  }
+  if (!confirmed) return;
+
+  // Now we populate the form with its new contents
+  let newDraftMarkdown;
+  let newDraftDelta;
+  if (draft.body) {
+    try {
+      newDraftDelta = JSON.parse(draft.body);
+    } catch (e) {
+      newDraftMarkdown = draft.body;
+    }
+  }
+  if (newDraftDelta && state.quillEditorState.markdownMode) {
+    state.quillEditorState.markdownMode = false;
+  } else if (newDraftMarkdown && !state.quillEditorState.markdownMode) {
+    state.quillEditorState.markdownMode = true;
+  }
+
+  if (newDraftDelta) {
+    state.quillEditorState.editor.setContents(newDraftDelta);
+  } else if (newDraftMarkdown) {
+    state.quillEditorState.editor.setText(newDraftMarkdown);
+  }
+  (titleInput as HTMLInputElement).value = draft.title;
+  state.form.title = draft.title;
+  state.activeTag = draft.tag;
+  state.fromDraft = draft.id;
+  m.redraw();
+};
+
+export const NewThreadForm: m.Component<{ header: boolean, isModal: boolean }, IState> = {
+  view: (vnode: VnodeDOM<{ header: boolean, isModal: boolean }, IState>) => {
+    const author = app.user.activeAccount;
+    const activeEntity = app.community ? app.community : app.chain;
+    const activeEntityInfo = app.community ? app.community.meta : app.chain.meta.chain;
+    if (vnode.state.quillEditorState?.container) vnode.state.quillEditorState.container.tabIndex = 8;
+
+    // init
+    if (vnode.state.form === undefined) vnode.state.form = {};
+    if (vnode.state.error === undefined) vnode.state.error = {};
+    if (vnode.state.uploadsInProgress === undefined) vnode.state.uploadsInProgress = 0;
+    if (vnode.state.newType === undefined) vnode.state.newType = 'Discussion';
+    const { error } = vnode.state;
+
+    const getUrlForLinkPost = _.debounce(async () => {
+      const res = await getLinkTitle(vnode.state.form.url);
+      if (res === '404: Not Found' || res === '500: Server Error') {
```

Comment:
This seems brittle, is there a way we avoid passing an error message and pass a status code instead?

Reply:
I don't quite follow?

Reply:
I don't quite follow?

Reply:
Oh, this is for _links_ that return a 404/500. We should come up with a better way to pass those from the backend but it's probably outside the scope of this PR, so never mind